packages/fs-detectors/src/packageManagers/package-managers.ts

```diff
@@ -0,0 +1,57 @@
+import type { Framework } from '@vercel/frameworks';
+
+export const packageManagers: Array<
+  Omit<Framework, 'description' | 'getOutputDirName' | 'settings'>
+> = [
+  {
+    name: 'npm',
+    slug: 'npm',
+    logo: '',
+    darkModeLogo: '',
+    detectors: {
+      some: [
+        {
+          path: 'package-lock.json',
+        },
+        {
+          path: 'package.json',
+          matchContent: '"packageManager":\\s*"npm@.*"',
+        },
+      ],
+    },
+  },
+  {
+    name: 'yarn',
+    slug: 'yarn',
+    logo: '',
+    darkModeLogo: '',
+    detectors: {
+      some: [
+        {
+          path: 'yarn.lock',
+        },
+        {
+          path: 'package.json',
+          matchContent: '"packageManager":\\s*"yarn@.*"',
+        },
+      ],
```

Comment:
you might want to add another rule here for when a repo doesnt have a lock file, and has a package file with no `packageManager` field set. 

```ts
{
  path: 'package.json'
}
```

(we default to yarn in this case)